File: ps/swea/2005.cpp
#include <iostream>
using namespace std;

int main() {
    int T;
    int n;
    int array[11] = {};
    int temp;
    int temp2;
    array[1] = 1;
    cin >> T;
    for (int i = 0; i < T; i++) {
        cin >> n; 
        cout << "#" << i+1 << " " << endl;
        for (int k =0; k < 11; k++) {
            array[k] = 0;
        }
        array[1] =1;
        for (int j = 1; j <= n; j++) {
            temp2 = 0;
            for (int k = 1; k <= j; k++) {
                temp = array[k];
                cout << array[k]+temp2 << " ";
                array[k] = array[k]+temp2;
                temp2 = temp;
            }
            cout << endl;
        }
    }
    return 0;
}

File: ps/swea/2005.py
T = int(input())
for tc in range(T):
    array = [0]*11
    array[1] = 1
    n = int(input())
    print(f"#{tc+1}")
    for i in range(1,n+1):
        temp2 = 0
        for k in range(1, i+1):
            temp = array[k]
            print(array[k] + temp2, end=" ")
            array[k], temp2 = array[k] + temp2, temp
        print()
